quantlib/instruments/credit_default_swap.pyx

```diff
@@ -178,7 +179,13 @@ cdef class CreditDefaultSwap(Instrument):
 
     @property
     def coupons(self):
-        return cashflow.leg_items(_get_cds(self).coupons())
+        cdef vector[shared_ptr[_cf.CashFlow]] coupons = _get_cds(self).coupons()
+        cdef vector[shared_ptr[_cf.CashFlow]].iterator it = coupons.begin()
+        cdef _cf.CashFlow* _thiscf
+        while it != coupons.end():
+            _thiscf = deref(it).get()
+            yield (_thiscf.amount(), _pydate_from_qldate(_thiscf.date()))
```

Comment:
Makes sense, I went a little overboard with the generators. How about this: I'll keep __iter__ as a generator and put back the other methods as they were.